triggers/simulations/overview.rst

```diff
@@ -0,0 +1,461 @@
+
+These performance metrics may then be subjected to an offline analysis.
+
+When architecting a collection of triggers and templates designed to interact with contracts on a ledger, a component based design is taken for describing the simulation. Each component models:
+
+- a composable set of primitive trigger definitions - this allows:
+
+  - complex trigger components to be modelled by composing individual abstraction layers (e.g. trigger initialization, scheduling of heartbeat messages, etc.)
+  - non-standard and unsupported trigger definitions to be constructed, tested and experimented with (e.g. filtering of received transaction events, off-ledger trigger communication, etc.).
+- the ledger
+- and external components that may interact with the ledger, but be otherwise hidden to trigger code
+
+An internal reporting component is responsible for collecting and storing simulation metric data for offline analysis. This data may be saved at the end of a simulation (in CSV files) for latter offline analysis.
+
+Trigger simulations are `Scalatests <https://www.scalatest.org>`_ that extend the trait ``TriggerMultiProcessSimulation``. In extending this trait, the key method that a developer needs to implement is the method:
+
+.. code-block:: scala
+  protected def triggerMultiProcessSimulation: Behavior[Unit]
+
+This method is used to define all the components that a given simulation is to take into account. So, to define a simulation with no trigger components and no external components, one could write:
+
+.. code-block:: scala
+  class ExampleSimulation extends TriggerMultiProcessSimulation {
+    override protected def triggerMultiProcessSimulation: Behavior[Unit] = {
+      implicit def applicationId: ApiTypes.ApplicationId = this.applicationId
+
+      Behaviors.setup { context =>
+        val setup = for {
+          client <- defaultLedgerClient()
+          party <- allocateParty(client)
+        } yield (client, party)
+        val (client, actAs) = Await.result(setup, simulationConfig.simulationSetupTimeout)
+        val ledger = context.spawn(LedgerProcess.create(client), "ledger")
+
+        // Trigger and external components could be defined here
+
+        Behaviors.empty
+      }
+    }
+  }
+
+Trigger simulations may have their default configurations modified by overriding the inherited field:
+
+.. code-block:: scala
+  protected implicit lazy val simulationConfig: TriggerSimulationConfig
+
+So, to have a simulation run for 42 seconds, one would override with:
+
+.. code-block:: scala
+  override protected implicit lazy val simulationConfig: TriggerSimulationConfig =
+    TriggerSimulationConfig(simulationDuration = 42.seconds)
+
+As trigger simulations are defined as Scalatests, they may be ran using a variant of ``bazel test`` (assuming the relevant ``BUILD.bazel`` files are also defined to allow the simulation code to build).
```

Comment:
`As` is overloaded here due to its different use later in the sentence, so I tend to opt for `Because`
`As` -> `Because`